src/Glpi/Controller/GenericFormController.php

```diff
@@ -0,0 +1,193 @@
+
+        if (!$object instanceof \CommonDBTM) {
+            Event::log(
+                '-',
+                \strtolower(\basename($class)),
+                $object::getLogLevel(),
+                $object::getLogServiceName(),
+                sprintf(__('%1$s tried to execute the "%2$s" action on the item %3$s, but this type does not support POST actions.'), $_SESSION["glpiname"], $form_action, $post_data["name"])
+            );
+
+            return null;
+        }
+
+        // Permissions
+        $object->check($id, self::ACTIONS_AND_CHECKS[$form_action]['permission'] ?? READ, $post_data);
+
+        // Special case for GET
+        if ($form_action === 'get' && $request->getMethod() === 'GET') {
+            return $this->displayForm($object, $request, $isTemplateForm, $class);
+        }
+
+        $result = $object->callFormAction($form_action, $post_data);
+
+        if ($result) {
+            Event::log(
+                $result,
+                \strtolower(\basename($class)),
+                $object::getLogLevel(),
+                $object::getLogServiceName(),
+                sprintf(__('%1$s executes the "%2$s" action on the item %3$s'), $_SESSION["glpiname"], $form_action, $post_data["name"])
+            );
+        }
+
+        // Specific case for "add"
+        if ($result && $form_action === 'add' && $_SESSION['glpibackcreated']) {
+            return new RedirectResponse($object->getLinkURL());
+        }
+
+        $post_action = $object::getPostFormAction($form_action) ?? 'list';
+        if (!\in_array($post_action, ['back', 'list', 'form'])) {
+            $post_action = 'list';
+        }
+
+        return match ($post_action) {
+            'back' => new RedirectResponse(Html::getBackUrl()),
+            'item' => new RedirectResponse(Toolbox::getItemTypeFormURL($class) . "?id=" . ($result ?: $id)),
+            'form' => new RedirectResponse($object::getFormURLWithID($id)),
+            'list' => new RedirectResponse($object->getRedirectToListUrl()),
+        };
+    }
+
+    /**
+     * @param class-string<CommonDBTM> $class
+     */
+    private function getCurrentAllowedAction(Request $request, string $class): ?string
+    {
+        if ($request->getMethod() === 'POST') {
+            foreach (\array_keys(self::ACTIONS_AND_CHECKS) as $action) {
+                if (
+                    $request->request->has($action)
+                    && \method_exists($class, $action)
+                ) {
+                    return $action;
+                }
+            }
+        }
+
+        return $request->getMethod() === 'GET' ? 'get' : null;
+    }
+
+    public function displayForm(CommonDBTM $object, Request $request, bool $isTemplateForm, string $class): Response
+    {
+        $form_options = $object->getFormOptionsFromUrl($request->query->all());
+        $form_options['withtemplate'] = $isTemplateForm;
```

Comment:
`withtemplate` is not a boolean, it may have the `1` or `2` value. Also, it should not be passed to the form if the corresponding objects does not support templating.

To reproduce the existing behaviour, you should probably use the following code:

```suggestion
        if ($object->maybeTemplate()) {
            $form_options['withtemplate'] = $request->query->get('withtemplate', '');
        }
```